src/main/java/org/example/jvspringbootfirstbook/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,173 @@
+package org.example.jvspringbootfirstbook.service.order;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.example.jvspringbootfirstbook.dto.order.OrderDto;
+import org.example.jvspringbootfirstbook.dto.order.OrderItemDto;
+import org.example.jvspringbootfirstbook.dto.order.PlacingOrderRequestDto;
+import org.example.jvspringbootfirstbook.exception.EntityNotFoundException;
+import org.example.jvspringbootfirstbook.mapper.OrderItemMapper;
+import org.example.jvspringbootfirstbook.mapper.OrderMapper;
+import org.example.jvspringbootfirstbook.model.CartItem;
+import org.example.jvspringbootfirstbook.model.Order;
+import org.example.jvspringbootfirstbook.model.OrderItem;
+import org.example.jvspringbootfirstbook.model.ShoppingCart;
+import org.example.jvspringbootfirstbook.model.Status;
+import org.example.jvspringbootfirstbook.model.User;
+import org.example.jvspringbootfirstbook.repository.cart.ShoppingCartRepository;
+import org.example.jvspringbootfirstbook.repository.order.OrderItemRepository;
+import org.example.jvspringbootfirstbook.repository.order.OrderRepository;
+import org.example.jvspringbootfirstbook.repository.user.UserRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderDto makeOrder(PlacingOrderRequestDto requestDto) {
+        User user = userRepository.findByShippingAddress(requestDto.shippingAddress())
+                .orElseThrow(
+                        () -> new EntityNotFoundException("No user found for shipping address"
```

Comment:
empty space at the end is missing

```suggestion
                        () -> new EntityNotFoundException("No user found for shipping address "
```